src/Amalgam/GeneralizedDistance.h

```diff
@@ -404,102 +457,76 @@ class GeneralizedDistance
 			return -std::numeric_limits<double>::infinity();
 	}
 
-	//computes the distance term for a nominal when two universally symmetric nominals are equal
-	__forceinline double ComputeDistanceTermNominalUniversallySymmetricExactMatch(size_t index, bool high_accuracy)
+	//computes the base of the difference between two nominal values that exactly match without exponentiation
+	__forceinline double ComputeDistanceTermNominalBaseExactMatchFromDeviation(size_t index, double deviation, bool high_accuracy)
 	{
-		if(!DoesFeatureHaveDeviation(index))
+		if(!DoesFeatureHaveDeviation(index) || computeSurprisal)
 			return 0.0;
 
-		double weight = featureParams[index].weight;
-		double deviation = featureParams[index].deviation;
-
-		//infinite pValues are treated the same as 1 for distance terms,
-		//and are the same value regardless of high_accuracy
-		if(pValue == 1 || pValue == std::numeric_limits<double>::infinity()
-				|| pValue == -std::numeric_limits<double>::infinity())
-			return deviation * weight;
-
-		if(pValue == 0)
-		{
-			if(high_accuracy)
-				return std::pow(deviation, weight);
-			else
-				return FastPow(deviation, weight);
-		}
-		else
-		{
-			if(high_accuracy)
-				return std::pow(deviation, pValue) * weight;
-			else
-				return FastPow(deviation, pValue) * weight;
-		}
+		return deviation;
```

Comment:
Just so I'm not missing something: this is just returning the deviation you send in? Except for that first if. Not sure that lines up with the comment? Or maybe I'm missing something...